src/metrics-key-builder.ts

```diff
@@ -38,7 +41,11 @@ export class MetricsKeyBuilder {
 
         let path = url.substring(idxStart, idxEnd);
 
-        return this.pathToKey(path);
+        let key: string = this.pathToKey(path);
+        if (statusCode) {
+            key = `${key}.${statusCode}`;
+        }
```

Comment:
`nit` this is duplicated in both methods, maybe it would be good to extract a function